src/rust/ensogl/src/display/style/path.rs

```diff
@@ -24,37 +24,42 @@ impl Path {
     pub fn from_segments<T,I,Item>(t:T) -> Self
     where T : IntoIterator<IntoIter=I,Item=Item>,
           I : DoubleEndedIterator + Iterator<Item=Item>,
-          Item : Into<String> {
+          Item : ToString {
         Self::from_rev_segments(t.into_iter().rev())
     }
 
     /// Builds the path from reversed segment iterator. See `from_segments` to learn more.
     pub fn from_rev_segments<T,Item>(t:T) -> Self
-    where T : IntoIterator<Item=Item>,
-          Item : Into<String> {
-        let rev_segments = t.into_iter().map(|s|s.into()).collect();
-        Self {rev_segments}
+    where T:IntoIterator<Item=Item>, Item:ToString {
+        Self { rev_segments : t.into_iter().map(|s|s.to_string()).collect() }
     }
-}
 
-impl From<&str> for Path {
-    fn from(t:&str) -> Self {
-        Self::from_rev_segments(t.rsplit('.'))
+    /// Empty path constructor.
+    pub fn empty() -> Self {
+        Self { rev_segments : default() }
     }
 }
 
-impl From<&Path> for Path {
-    fn from(t:&Path) -> Self {
-        t.clone()
+impl AsRef<Path> for Path { fn as_ref(&self) -> &Path { self } }
```

Comment:
I suspect this is required for some mysterious reasons, and will help if you share them in a comment :)

Reply:
As agreed via phone, no comment needed here